developers/wallets/exchanges.md

```diff
@@ -0,0 +1,174 @@
+primitives. Either party to an exchange is capable of requesting Verifiable
+Presentations and of providing one or more Verifiable Credentials that might be
+necessary to establish trust and/or gain authorization capabilities, and either
+party is capable of presenting credentials that they hold or that they have
+issued. Specific workflows can be configured to expect specific presentations
+and credentials and to reject deviations from the expected flow of information.
+
+When a workflow service determines that a particular message is not acceptable,
+it raises an error by responding with a `4xx` HTTP status message and a JSON
+object that expresses information about the error.
+
+Below is an example of a typical exchange:
+
+```mermaid
+sequenceDiagram
+    participant H as Holder
+    participant W as Holder Coordinator (Wallet)
+    participant I as Issuer/Verifier Coordinator
+    autonumber
+    Note right of H: Start exchange
+    W->>I: Initiate
+    Note right of W: POST /workflows/123/exchanges/123 &mdash; HTTP request to start exchange (e.g., send credentials, get credentials)
+    I->>W: Verifiable Presentation Request (VPR)
+    Note left of I: VPR includes method of interaction, for purposes of exchange
+    W->>I: Verifiable Presentation (VP)
+    Note right of W: POST /workflows/123/exchanges/abc &mdash; sent via interaction mechanism to meet requirements of exchange
+    I->>W: Verifiable Presentation
+    Note left of I: VP includes result of exchange (e.g., VCs), or VPR with new interaction request, or error description
+```
+
+The exchange client (Wallet) code for a flow like the above may look similar to
+the following:
+
+```js
+const receivedExchangeUrl = 'https://vcapi.example.com/workflows/123/exchanges/123';
+
+const response = await fetch(receivedExchangeUrl, {
+  method: 'POST'
+  body: JSON.stringify({})
+});
+const body = response.json();
+
+function checkResponse(body) {
+  if('verifiablePresentationRequest' in body) {
+    // use the information in the Verifiable Presentation Request to find a
+    // credentail that fullfills the request, then send a Verifiable
+    // Presentation as the response to the interaction endpoint included in the
+    // Verifiable Presentation Request
+    const presentation = findCredentialAndCreatePresentation();
+    // TODO: properly extract interaction URL
+    const interactionUrl = body.interact.service[0].serviceEndpoint;
+    const response = await fetch(interactionUrl, {
+      method: 'POST',
+      body: JSON.stringify(presentation)
+    });
+    checkResponse(response.json());
+  }
+  // TODO: should these be exclusive?
+  if('verifiablePresentation' in body) {
+    // the Wallet has received a Verifiable Presentation containing one or more
+    // Verifiable Credentials--validate and verify them per your use case
+  }
+  if('redirectUrl' in body) {
+    // TODO: take the user to the new location OR does the exchange "move"?
```

Comment:
Exchange does not "move", the user is just taken elsewhere for any other kind of continuing interaction with the coordinator.

```suggestion
    // take the user to the new location
```